### PR TITLE
Allow sending additional GELF fields in logs

### PR DESCRIPTION
**Pull request by rtxm**

### CODE REVIEW

**Review by merwan — APPROVED** [review]
LGTM
A link to the [GELF payload specification](https://docs.graylog.org/en/4.0/pages/gelf.html#gelf-payload-specification) in the description would have helped understand why we need to prefix additional fields with an underscore
>every field you send and prefix with an underscore (_) will be treated as an additional field. Allowed characters in field names are any word character (letter, number, underscore), dashes and dots. The verifying regular expression is: ^[\w\.\-]*$. Libraries SHOULD not allow to send id as additional field (_id). Graylog server nodes omit this field automatically.

That being said, ideally we should prevent from sending an `id` field too.

**Comment by merwan** [comment]
There is also some copy/paste for the log handling that could maybe be extracted to a common logging formatting module/function/whatever

**Comment by rtxm** [comment]
copy / paste fixed @merwan

File: src/eventail/gelf.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            "level": criticity,
+            "_levelname": level_name,
+            "host": f"{endpoint.logical_service}@{endpoint.HOSTNAME}",
+            "timestamp": time.time(),
+            "_conversation_id": conversation_id,
+            "_logical_service": endpoint.logical_service,
+            "_worker_pid": endpoint.ID,
+        }
+        for key, value in additional_fields.items():
+            assert key != "id" and key != "_id", "GELF: 'id' can't be used as key name!"
```
Comment (merwan) [inline src/eventail/gelf.py:39]: Should we really fail in such case? What about gently ignoring those keys and log everything else.

Comment (rtxm) [reply src/eventail/gelf.py:39]: it's like passing the wrong keyword parameter to a function.

Comment (merwan) [reply src/eventail/gelf.py:39]: What will happen to a running service when this assertion fails? As long as the failure is visible by the developer and nothing is lost, then it's good for me

Comment (rtxm) [reply src/eventail/gelf.py:39]: If it is properly tested, the service won't reach production with this kind of error. I see that like a programming error, not a data error.

**Review by merwan — APPROVED** [review]

**Pull request merged**